optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -625,4 +643,231 @@ private boolean shouldIgnoreMember(ClassInfo declaringClass) {
+            CustomPhaseConfig customPhaseConfig = (CustomPhaseConfig) phaseConfig;
+            reflectiveClassSet.addAll(customPhaseConfig.getCustomPhaseCommandClassList());
+        } else if (phaseConfig instanceof ConstructionHeuristicPhaseConfig) {
+            ConstructionHeuristicPhaseConfig chPhaseConfig = (ConstructionHeuristicPhaseConfig) phaseConfig;
+            if (chPhaseConfig.getMoveSelectorConfigList() != null) {
+                for (MoveSelectorConfig<?> moveSelectorConfig : chPhaseConfig.getMoveSelectorConfigList()) {
+                    registerMoveSelectorConfig(moveSelectorConfig, reflectiveClassSet);
+                }
+            }
+            registerEntityPlacerConfig(chPhaseConfig.getEntityPlacerConfig(), reflectiveClassSet);
+        } else if (phaseConfig instanceof LocalSearchPhaseConfig) {
+            LocalSearchPhaseConfig lsPhaseConfig = (LocalSearchPhaseConfig) phaseConfig;
+            registerMoveSelectorConfig(lsPhaseConfig.getMoveSelectorConfig(), reflectiveClassSet);
+        } else if (phaseConfig instanceof ExhaustiveSearchPhaseConfig) {
+            ExhaustiveSearchPhaseConfig esPhaseConfig = (ExhaustiveSearchPhaseConfig) phaseConfig;
+            registerMoveSelectorConfig(esPhaseConfig.getMoveSelectorConfig(), reflectiveClassSet);
+            registerEntitySelectorConfig(esPhaseConfig.getEntitySelectorConfig(), reflectiveClassSet);
+        } else if (phaseConfig instanceof PartitionedSearchPhaseConfig) {
+            PartitionedSearchPhaseConfig psPhaseConfig = (PartitionedSearchPhaseConfig) phaseConfig;
+            registerClassForReflection(psPhaseConfig.getSolutionPartitionerClass(), reflectiveClassSet);
+            if (psPhaseConfig.getPhaseConfigList() != null) {
+                psPhaseConfig.getPhaseConfigList().forEach(pc -> registerPhaseConfigForReflection(pc, reflectiveClassSet));
+            }
+        } else if (phaseConfig instanceof NoChangePhaseConfig) {
+            // Need to do nothing, since termination config is registered above
+        } else {
+            throw new IllegalStateException("Unhandled phase config class (" + phaseConfig.getClass() + ").");
+        }
+    }
+
+    private void registerMoveSelectorConfig(MoveSelectorConfig<?> moveSelectorConfig, Set<Class<?>> reflectiveClassSet) {
+        if (moveSelectorConfig == null) {
+            return;
+        }
+
+        registerClassForReflection(moveSelectorConfig.getFilterClass(), reflectiveClassSet);
+        registerClassForReflection(moveSelectorConfig.getSorterClass(), reflectiveClassSet);
+        registerClassForReflection(moveSelectorConfig.getSorterComparatorClass(), reflectiveClassSet);
+        registerClassForReflection(moveSelectorConfig.getSorterWeightFactoryClass(), reflectiveClassSet);
+        registerClassForReflection(moveSelectorConfig.getProbabilityWeightFactoryClass(), reflectiveClassSet);
+
+        if (moveSelectorConfig instanceof AbstractPillarMoveSelectorConfig) {
```

Comment:
Ok, this is code in optaplanner-quarkus is getting know way too much from MoveSelectorConfig (it's internal structure etc) than is good for it. When MoveSelectorConfig refactors, this code needs to refactor along. When a new MoveSelectorConfig is create, this code needs to add it too (and it's easy to forget it).

What if we add a method on MoveSelectorConfig called `retrieveCustomClassesForReflection()`? In fact, we should add it on PhaseConfig. No, on SolverConfig which then calls it as needed down

Reply:
Created method `void visitReferencedClasses(Consumer<Class<?>> classVisitor)` in `AbstractConfig`. Decided to make it a visitor since it make the code cleaner (do not need to do null checks on classes)